src/editor.c

```diff
@@ -2045,8 +2114,13 @@ gboolean editor_show_calltip(GeanyEditor *editor, gint pos)
 	editor_find_current_word(editor, pos - 1, word, sizeof word, NULL);
 	if (word[0] == '\0')
 		return FALSE;
-
-	str = find_calltip(word, editor->document->file_type);
+	
+	scope[0] = '\0';
+	//const gchar *sname;
+	//symbols_get_current_scope(editor->document, &sname);
+	//printf("Current scope: %s\n", sname);
+	read_scope_prefix(editor, pos - 1, scope, sizeof scope, NULL);
+	str = find_calltip(word, scope, editor->document->file_type);
```

Comment:
`symbols_get_current_scope()` returns a somewhat limited view of the lexical scope IIUC, so its not what you want until you need to resolve the highest level name in your prefix.